bft_types/src/lib.rs

```diff
@@ -154,39 +261,70 @@ mod tests {
-        assert_eq!(Instruction::from_char('>'), Some(Instruction::MoveRight));
-        assert_eq!(Instruction::from_char('+'), Some(Instruction::Increment));
-        assert_eq!(Instruction::from_char('-'), Some(Instruction::Decrement));
-        assert_eq!(Instruction::from_char(','), Some(Instruction::Input));
-        assert_eq!(Instruction::from_char('.'), Some(Instruction::Output));
+        assert_eq!(Instruction::from_char('<'), Some(Instruction::MoveLeft));
+        assert_eq!(Instruction::from_char('>'), Some(Instruction::MoveRight));
+        assert_eq!(Instruction::from_char('+'), Some(Instruction::Increment));
+        assert_eq!(Instruction::from_char('-'), Some(Instruction::Decrement));
+        assert_eq!(Instruction::from_char(','), Some(Instruction::Input));
+        assert_eq!(Instruction::from_char('.'), Some(Instruction::Output));
         assert_eq!(
-            Instruction::from_char('['),
-            Some(Instruction::ConditionalJumpForward)
+            Instruction::from_char('['),
+            Some(Instruction::ConditionalJumpForward)
         );
         assert_eq!(
-            Instruction::from_char(']'),
-            Some(Instruction::ConditionalJumpBackward)
+            Instruction::from_char(']'),
+            Some(Instruction::ConditionalJumpBackward)
         );
     }
 
     /// Check that a program can be constructed and records line and column numbers correctly
     #[test]
     fn parse_program() {
-        let filename = Path::new("test_file.bf");
+        let filename = Path::new("test_file.bf");
         let lines = "_<\n__<\n";
-        let placeholder_instruction_type = Instruction::from_char('<').unwrap(); // probably shouldn't use unwrap here but I'm getting fed up of this and it'll do for now
+        let placeholder_instruction_type = Instruction::from_char('<').unwrap(); // probably shouldn't use unwrap here but I'm getting fed up of this and it'll do for now
 
         let bf_program = BfProgram::new(filename, lines);
 
         assert_eq!(bf_program.name(), filename);
 
-        let expected_instruction =
-            LocalisedInstruction::new(placeholder_instruction_type.clone(), 0, 1);
-        assert_eq!(bf_program.instructions[0], expected_instruction);
+        let expected_instruction =
+            LocalisedInstruction::new(placeholder_instruction_type.clone(), 1, 2);
+        assert_eq!(bf_program.instructions.get(0), Some(&expected_instruction));
 
-        let expected_instruction =
-            LocalisedInstruction::new(placeholder_instruction_type.clone(), 1, 2);
+        let expected_instruction =
+            LocalisedInstruction::new(placeholder_instruction_type.clone(), 2, 3);
         assert_eq!(bf_program.instructions.get(1), Some(&expected_instruction));
     }
+
+    /// check that analysing a valid program works
+    #[test]
+    fn test_analyse_good() {
+        let filename = Path::new("test_file.bf");
```

Comment:
If you're going to include test cases, it'd be nice if they weren't:

1. stored directly in the workspace root
2. referenced without some use of an appropriate environment variable - if I were inside the `bft_types` directory and ran `cargo test` would this work?